proposals/004-skips/README.md

```diff
@@ -0,0 +1,179 @@
+
+[Dapr]: (https://dapr.io/)
+[Envoy]: (https://www.envoyproxy.io/)
+
+### Existing applications
+
+Here are some existing prototypes that I have found that runs Wasm workloads side by side with Linux containers:
+
+- [SpinKube - the first look at WebAssembly/WASI application (SpinApp) on Kubernetes](https://dev.to/thangchung/spinkube-the-first-look-at-webassemblywasi-application-spinapp-on-kubernetes-36jd) and [Coffeeshop repo](https://github.com/thangchung/coffeeshop-on-spinkube/tree/feat/global-azure-2024)
+- [Istio Wasm Demo](https://github.com/keithmattix/istio-wasm-demo)
+
+
+## Proposal
+
+
+The main changes are
+
+1. A new field `InitContainers` in the `spinapp_types.go` that defines a list of sidecar containers to be included in the Deployment.
+```go
+// InitContainers defines the list of sidecar containers to be included in the deployment.
+//
+// These containers will not include the main Spin App. They share the Spin App's
+// environment variables and volumes.
+// +kubebuilder:validation:Optional
+InitContainers []corev1.Container `json:"containers,omitempty"`
+```
+
+2. Adds containers to the deployment creation in `spinapp_controller.go`:
+```go
+var containers []corev1.Container
+if len(app.Spec.InitContainers) > 0 {
+    for _, c := range app.Spec.InitContainers {
+        if c.Image == app.Spec.Image {
+            return nil, errors.New("container in app.Spec.InitContainers must have a different image than Spin App")
+        }
+        if c.Name == "" {
+            return nil, errors.New("container in app.Spec.InitContainers must have a name")
+        }
+        if c.Name == app.Name {
+            return nil, errors.New("container in app.Spec.InitContainers must have a different name than the Spin App")
+        }
+        c.Env = append(c.Env, env...)
+        c.VolumeMounts = append(c.VolumeMounts, volumeMounts...)
+        if c.Resources.Limits == nil && c.Resources.Requests == nil {
+            c.Resources = resources
+        }
+        if c.LivenessProbe == nil {
+            c.LivenessProbe = livenessProbe
+        }
+        if c.ReadinessProbe == nil {
+            c.ReadinessProbe = readinessProbe
+        }
+        containers = append(containers, c)
+    }
+}
+```
+
+See the full diff [here](https://github.com/spinkube/spin-operator/compare/main...Mossaka:spin-operator-msk:sidecars)
+
+[sidecar containers]: (https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/)
+[InitContainers]: (https://kubernetes.io/docs/concepts/workloads/pods/init-containers/)
+
+### Ports
+
+Sidecar containers must avoid using port 80, which is reserved for the Spin application. Other ports like 8080 or 8081 can be used.
+
+### Services
+
+Sidecar containers can communicate internally using `localhost`, but if external access is needed, users must create a separate service for the sidecar container.
+
+### Restrictions
+
+* Sidecar containers cannot use the same image or name as the Spin App. 
```

Comment:
Can sidecar containers be wasm applications or only Linux images? Do we not care?